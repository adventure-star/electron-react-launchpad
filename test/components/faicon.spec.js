import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import ReactTestRenderer from 'react-test-renderer';
import FaIcon from '../../src/renderer/components/faicon.jsx'

function setupStandard() {
  const standardRenderer = new ReactShallowRenderer();

  standardRenderer.render(<FaIcon code='home' color='#fff' className='testClass' />);
  const component = standardRenderer.getRenderOutput();

  return {
    component
  };
}


function setupSpan() {
  const spanRenderer = ReactTestRenderer.create(<FaIcon code='home' color='#fff' size={2} container='span' />);
  const spanComponent = spanRenderer.toJSON();

  return {
    spanRenderer,
    spanComponent
  };
}

describe('FaIcon component', () => {
  it('should be of type i', () => {
    const { component } = setupStandard();
    expect(component.type).toBe('i');
  });

  it('should contain no child elements (text)', () => {
    const { component } = setupStandard();
    expect(component.props.children).toBe(undefined);
  });

  it('classes should be "fa fa-home testClass"', () => {
    const { component } = setupStandard();
    expect(component.props.className).toBe("fa fa-home testClass");
  });

  it('color should be #fff', () => {
    const { component } = setupStandard();
    expect(component.props.style.color).toBe('#fff');
  });

  it('should be of type span', () => {
    const { spanComponent } = setupSpan();
    expect(spanComponent.type).toBe('span');
  });

  it('span should contain no child elements (text)', () => {
    const { spanComponent } = setupSpan();
    expect(spanComponent.props.children).toBe(undefined);
  });

  it('span classes should be "fa fa-home fa-2x"', () => {
    const { spanComponent } = setupSpan();
    expect(spanComponent.props.className).toBe("fa fa-home fa-2x");
  });

  it('Matches/Renders a snapshot', () => {
    const { component } = setupStandard();
    expect(component).toMatchSnapshot();

    const { spanComponent } = setupSpan();
    expect(spanComponent).toMatchSnapshot();
  });
});
