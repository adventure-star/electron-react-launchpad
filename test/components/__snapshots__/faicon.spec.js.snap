// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaIcon component Matches/Renders a snapshot 1`] = `
<i
  className="fa fa-home testClass"
  style={
    Object {
      "color": "#fff",
    }
  }
/>
`;

exports[`FaIcon component Matches/Renders a snapshot 2`] = `
<span
  className="fa fa-home fa-2x"
  style={
    Object {
      "color": "#fff",
    }
  }
/>
`;
